src/messaging/ExchangeMgr.cpp

```diff
@@ -221,16 +220,13 @@ void ExchangeManager::DispatchMessage(const PacketHeader & packetHeader, const P
     // If we found a handler or we need to create a new exchange context (EC).
     if (matchingUMH != nullptr)
     {
-        ExchangeContext::MessageReceiveFunct umhandler = nullptr;
-
-        ec = AllocContext(payloadHeader.GetExchangeID(), packetHeader.GetSourceNodeId().Value(), false, matchingUMH->AppState);
+        ec = AllocContext(payloadHeader.GetExchangeID(), packetHeader.GetSourceNodeId().Value(), false, matchingUMH->App);
         VerifyOrExit(ec != nullptr, err = CHIP_ERROR_NO_MEMORY);
 
-        umhandler = matchingUMH->Handler;
-
-        ChipLogProgress(ExchangeManager, "ec id: %d, AppState: 0x%x", (ec - ContextPool + 1), ec->GetAppState());
+        ChipLogProgress(ExchangeManager, "ec id: %d, Delegate: 0x%x", (ec - ContextPool + 1), ec->GetDelegate());
 
-        ec->HandleMessage(packetHeader, payloadHeader, msgBuf, umhandler);
+        ec->SetDelegate(matchingUMH->App);
```

Comment:
has the delegate already been set within AllocContext?

Reply:
Good catch